XenModel/Actions/Updates/DownloadAndUpdateClientAction.cs

```diff
@@ -181,35 +184,56 @@ protected override void Run()
                 if (Cancelling)
                     throw new CancelledException();
             }
+        }
 
-            ValidateMsi();
-
-            if (!File.Exists(outputPathAndFileName))
+        public void InstallMsi()
+        {
+            if (!File.Exists(_outputPathAndFileName))
                 throw new Exception(Messages.DOWNLOAD_CLIENT_INSTALLER_MSI_NOT_FOUND);
 
+            // Install the msi            
+            try
+            {
+                ValidateMsi();
+                // Start the install process, it will handle closing of application.
+                Process.Start(_outputPathAndFileName);
+                log.DebugFormat("Update {0} found and install started", _updateName);
+            }
+            catch (Exception e)
+            {
+                if (File.Exists(_outputPathAndFileName))
+                    File.Delete(_outputPathAndFileName);
+
+                log.Error("Exception occurred when starting the installation process.", e);
+                throw;
+            }
+            finally
+            {
+                MsiStream?.Dispose();
+            }
+
             Description = Messages.COMPLETED;
         }
 
         private void ValidateMsi()
         {
-            using (FileStream stream = new FileStream(outputPathAndFileName, FileMode.Open, FileAccess.Read))
-            {
-                var calculatedChecksum = string.Empty; 
+            MsiStream = new FileStream(_outputPathAndFileName, FileMode.Open, FileAccess.Read);
 
-                var hash = StreamUtilities.ComputeHash(stream, out _);
-                if (hash != null)
-                    calculatedChecksum = string.Join("", hash.Select(b => $"{b:x2}"));
+            var calculatedChecksum = string.Empty; 
 
-                // Check if calculatedChecksum matches what is in chcupdates.xml
-                if (!checksum.Equals(calculatedChecksum, StringComparison.InvariantCultureIgnoreCase))
-                    throw new Exception(Messages.UPDATE_CLIENT_INVALID_CHECKSUM );
-            }
+            var hash = StreamUtilities.ComputeHash(MsiStream, out _);
+            if (hash != null)
+                calculatedChecksum = string.Join("", hash.Select(b => $"{b:x2}"));
+
+            // Check if calculatedChecksum matches what is in chcupdates.xml
+            if (!_checksum.Equals(calculatedChecksum, StringComparison.InvariantCultureIgnoreCase))
+                throw new Exception(Messages.UPDATE_CLIENT_INVALID_CHECKSUM );
 
             bool valid = false;

```

Comment:
_**very**_ nit: you don't need to initialise `bool valid = false`. `bool valid;` works too since you're not using the `false` value in any execution path.

Reply:
While i dont need to I find it helps (mostly me) see what default values are rather than _knowing_ what they are when looking at code